bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -667,7 +685,21 @@ private boolean updateRollerStatus(ShellySettingsStatus status, @Nullable Shelly
             getThing().postEvent(SHELLY_EVENT_ROLLER_CALIB, false);
         }
 
-        return updateChannels ? ShellyComponents.updateRoller((ShellyBaseHandler) getThing(), rs, cs.id) : false;
+        return updateChannels ? ShellyComponents.updateRoller((ShellyBaseHandler) getThing(), rs, rIdx) : false;
+    }
+
+    private int getRollerIdx(ShellyDeviceProfile profile, @Nullable Integer id) {
+        List<ShellySettingsRoller> rollers = profile.settings.rollers;
+        if (id != null && rollers != null) {
+            int idx = 0;
+            for (ShellySettingsRoller roller : rollers) {
+                if (roller.isValid && roller.id != null && roller.id.intValue() == id.intValue()) {
```

Comment:
I guess it is unexpected not finding the id and it will be okay for the binding to stop working in this scenario?

Reply:
yes, the id should be found, otherwise I can't map the status update to the right roller